cmd/experimental/kjobctl/pkg/builder/slurm_builder.go

```diff
@@ -179,33 +182,37 @@ func (b *slurmBuilder) build(ctx context.Context) ([]runtime.Object, error) {
 		return nil, err
 	}
 
+	objectMeta := b.buildObjectMeta(template.ObjectMeta)
+	objectMeta.GenerateName = ""
+	objectMeta.Name = b.objectName
+
 	job := &batchv1.Job{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Job",
-			APIVersion: "batch/v1",
-		},
-		ObjectMeta: b.buildObjectMeta(template.ObjectMeta),
+		TypeMeta:   metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+		ObjectMeta: objectMeta,
 		Spec:       template.Template.Spec,
 	}
 	job.Spec.CompletionMode = ptr.To(batchv1.IndexedCompletion)
-
-	objectName := b.generatePrefixName() + utilrand.String(5)
-	job.ObjectMeta.GenerateName = ""
-	job.ObjectMeta.Name = objectName
+	job.Spec.Template.Spec.Subdomain = b.objectName
 
 	configMap := &corev1.ConfigMap{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "ConfigMap",
-			APIVersion: "v1",
-		},
-		ObjectMeta: b.buildObjectMeta(template.ObjectMeta),
+		TypeMeta:   metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+		ObjectMeta: objectMeta,
 		Data: map[string]string{
 			slurmEntrypointFilename: b.buildEntrypointScript(),
 			slurmScriptFilename:     b.scriptContent,
 		},
 	}
-	configMap.ObjectMeta.GenerateName = ""
-	configMap.ObjectMeta.Name = objectName
+
+	service := &corev1.Service{
```

Comment:
How is it ensured that the service is removed once the job is deleted?

Reply:
We can't remove it automatically. The same situation with ConfigMap. But we can delete it using `kjobctl delete` command.

Reply:
What about setting owner reference?

Reply:
Cool, thanks!